tcparse/templates/stcmd_default.cmd

```diff
@@ -57,18 +57,23 @@ asynSetTraceInfoMask("$(ASYN_PORT)", -1, 5)
 
 {% for motor in motors %}
 epicsEnvSet("AXISCONFIG",      "{{motor.axisconfig}}")
-epicsEnvSet("MOTOR_NAME",      "{{motor.name}}")
 epicsEnvSet("AXIS_NO",         "{{motor.axis_no}}")
 epicsEnvSet("DESC",            "{{motor.desc}}")
 epicsEnvSet("EGU",             "{{motor.egu}}")
 epicsEnvSet("PREC",            "{{motor.prec}}")
 epicsEnvSet("ECAXISFIELDINIT", "{{motor.additional_fields}}")
 epicsEnvSet("AMPLIFIER_FLAGS", "{{motor.amplifier_flags}}")
+epicsEnvSet("MOTOR_PREFIX",    "{{motor.name[0]}}")
+epicsEnvSet("MOTOR_NAME",      "{{motor.name[1]}}")
 
 EthercatMCCreateAxis("$(MOTOR_PORT)", "$(AXIS_NO)", "$(AMPLIFIER_FLAGS)", "$(AXISCONFIG)")
-dbLoadRecords("EthercatMC.template", "PREFIX=$(PREFIX), MOTOR_NAME=$(MOTOR_NAME), R=$(MOTOR_NAME)-, MOTOR_PORT=$(MOTOR_PORT), ASYN_PORT=$(ASYN_PORT), AXIS_NO=$(AXIS_NO), DESC=$(DESC), PREC=$(PREC) $(ECAXISFIELDINIT)")
-dbLoadRecords("EthercatMCreadback.template", "PREFIX=$(PREFIX), MOTOR_NAME=$(MOTOR_NAME), R=$(MOTOR_NAME)-, MOTOR_PORT=$(MOTOR_PORT), ASYN_PORT=$(ASYN_PORT), AXIS_NO=$(AXIS_NO), DESC=$(DESC), PREC=$(PREC) ")
-dbLoadRecords("EthercatMCdebug.template", "PREFIX=$(PREFIX), MOTOR_NAME=$(MOTOR_NAME), MOTOR_PORT=$(MOTOR_PORT), AXIS_NO=$(AXIS_NO), PREC=3")
+dbLoadRecords("EthercatMC.template", "PREFIX=$(MOTOR_PREFIX), MOTOR_NAME=$(MOTOR_NAME), R=$(MOTOR_NAME)-, MOTOR_PORT=$(MOTOR_PORT), ASYN_PORT=$(ASYN_PORT), AXIS_NO=$(AXIS_NO), DESC=$(DESC), PREC=$(PREC) $(ECAXISFIELDINIT)")
```

Comment:
Last I remember, the ECAXISFIELDINIT here was breaking my IOC. Has something changed?

Reply:
You're correct in remembering - but the issue was that it was unset. It's now set here, and configurable via `pytmc` pragmas.